### ISSUE TITLE
BSON arrays: MongoDB\BSON\toPHP & MongoDB\BSON\fromJson 

### ISSUE THREAD

**Issue by AntoineEsteve** [mongodb/mongo-php-driver#603]
### Description

**JSON extended arrays are incorrectly converted to PHP arrays when they contains BSON types.** 
Array indexes seems doubled.

### Environment
`mongodb/mongodb                       1.1.2              MongoDB driver library`
```
hhvm --version
HipHop VM 3.18.2 (rel)
Compiler: tags/HHVM-3.18.2-0-g1e28cfedefb439554cae05f1721bd6ea60c31839
Repo schema: 7aa473404e83b5cd2d1a53c6f3200a3a9012f2f1
```
_I know you are not maintaining HHVM anymore but there is probably the same problem with PHP_

### Tests

```
var_dump(\MongoDB\BSON\toPHP(\MongoDB\BSON\fromJson('{"myArray": [{"$oid": "2017-06-13T11:21:26.906Z"}]}')));

object(stdClass)#2 (1) {
  ["myArray"]=>
  array(1) {
    [1]=>
    object(MongoDB\BSON\ObjectID)#1 (1) {
      ["oid"]=>
      string(24) "201700601301102102609060"
    }
  }
}
```
**Expected behavior:** array indexes should be 0 instead of 1 

```
var_dump(\MongoDB\BSON\toPHP(\MongoDB\BSON\fromJson('[1,2,3,{"$date": "2017-06-13T11:21:26.906Z"}]')));
object(stdClass)#4 (4) {
  [0]=>
  int(1)
  [1]=>
  int(2)
  [2]=>
  int(3)
  [4]=>
  object(MongoDB\BSON\UTCDateTime)#3 (1) {
    ["milliseconds"]=>
    string(13) "1497352886906"
  }
}
```
**Expected behavior:** array indexes should be 0, 1, 2, 3 instead of 0, 1, 2, 4 

```
var_dump(\MongoDB\BSON\toPHP(\MongoDB\BSON\fromJson('[{"$oid": "55f2b3f1f657b3fa97c9c0a2"},{"$date": "2017-06-13T11:21:26.906Z"}]')));
object(stdClass)#6 (2) {
  [1]=>
  object(MongoDB\BSON\ObjectID)#4 (1) {
    ["oid"]=>
    string(24) "55f2b3f1f657b3fa97c9c0a2"
  }
  [3]=>
  object(MongoDB\BSON\UTCDateTime)#5 (1) {
    ["milliseconds"]=>
    string(13) "1497352886906"
  }
}
```
**Expected behavior:** array indexes should be 0, 1 instead of 1, 3

**Note:** Using a typemap do not help if you need to use the php array directly.
But if you use the typemap `'array' => '\MongoDB\Model\BSONArray'` and if you only use that method to inserts documents they will correctly be inserted.

**Comment by jmikola** [comment]
The first example doesn't appear to reproduce in version 1.2.9 of the PHP extension, but I can reproduce the second and third issues. I've opened [PHPC-974](https://jira.mongodb.org/browse/PHPC-974) to track this, so please watch that issue for updates.

**Comment by jmikola** [comment]
Upon further investigation, this looks like a bug in libbson 1.5.x ([CDRIVER-994](https://jira.mongodb.org/browse/CDRIVER-994)), which was fixed in the 1.6.0 release. The upcoming 1.3.0 release of the PHP extension bumps our libbson dependency, so this will be fixed there.

As for HHVM, you may find this resolved in the master branch of [mongo-hhvm-driver](https://github.com/mongodb/mongo-hhvm-driver), since that is essentially 1.3.0-dev and includes most of the same changes currently in the PHP extension's master branch.
